src/main/java/com/github/pedrovgs/androidwifiadb/adb/ADB.java

```diff
@@ -53,15 +55,16 @@ public boolean isInstalled() {
   }
 
   private boolean connectDeviceByIp(Device device) {
-    String deviceIp = getDeviceIp(device);
+    String deviceIp = device.getIp().length() > 0 ? device.getIp() : getDeviceIp(device);
```

Comment:
If I'm not wrong ``device.getIp()`` returns an String and that class contains a ``isEmpty`` method we can use instead of ``device.getIp().length() > 0``.